src/Adapter/Product/Combination/Repository/CombinationMultiShopRepository.php

```diff
@@ -0,0 +1,670 @@
+        ;
+
+        $result = $qb->execute()->fetchAssociative();
+
+        if (!$result) {
+            return null;
+        }
+
+        return new CombinationId((int) $result['id_product_attribute']);
+    }
+
+    /**
+     * Check if combination is associated with certain shop
+     *
+     * @param CombinationId $combinationId
+     * @param ShopId $shopId
+     *
+     * @return bool
+     */
+    public function isAssociatedWithShop(CombinationId $combinationId, ShopId $shopId): bool
+    {
+        $qb = $this->connection->createQueryBuilder()
+            ->select('pas.id_product_attribute')
+            ->from($this->dbPrefix . 'product_attribute_shop', 'pas')
+            ->where('pas.id_product_attribute = :combinationId')
+            ->andWhere('pas.id_shop = :shopId')
+            ->setParameter('combinationId', $combinationId->getValue())
+            ->setParameter('shopId', $shopId->getValue())
+        ;
+
+        $result = $qb->execute()->fetchAssociative();
+
+        return isset($result['id_product_attribute']);
+    }
+
+    /**
+     * Returns default combination ID identified as such in DB by default_on property
+     *
+     * @param ProductId $productId
+     * @param ShopId $shopId
+     *
+     * @return CombinationId|null
+     */
+    public function findDefaultCombinationIdForShop(ProductId $productId, ShopId $shopId): ?CombinationId
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('pas.id_product_attribute')
+            ->from($this->dbPrefix . 'product_attribute_shop', 'pas')
+            ->where('pas.id_product = :productId')
+            ->andWhere('pas.id_shop = :shopId')
+            ->andWhere('pas.default_on = 1')
+            ->addOrderBy('pas.id_product_attribute', 'ASC')
+            ->setParameter('productId', $productId->getValue())
+            ->setParameter('shopId', $shopId->getValue())
+        ;
+
+        $result = $qb->execute()->fetchAssociative();
+        if (empty($result['id_product_attribute'])) {
+            return null;
+        }
+
+        return new CombinationId((int) $result['id_product_attribute']);
+    }
+
+    /**
+     * @param CombinationId $combinationId
+     *
+     * @return ShopId[]
+     *
+     * @throws Exception
+     * @throws ShopException
+     */
+    public function getAssociatedShopIds(CombinationId $combinationId): array
+    {
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('id_shop')
+            ->from($this->dbPrefix . 'product_attribute_shop')
+            ->where('id_product_attribute = :combinationId')
+            ->setParameter('combinationId', $combinationId->getValue())
+        ;
+
+        $result = $qb->execute()->fetchAll();
+        if (empty($result)) {
+            return [];
+        }
+
+        $shops = [];
+        foreach ($result as $shop) {
+            $shops[] = new ShopId((int) $shop['id_shop']);
+        }
+
+        return $shops;
```

Comment:
It's not blocking but I've seen this several times and I really think that we should use array map more.
```suggestion
        return array_map(
            function (array $shop): ShopId {
                return new ShopId((int) $shop['id_shop']);
            },
            $qb->execute()->fetchAll()
        );
```